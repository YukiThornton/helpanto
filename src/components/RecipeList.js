import React from 'react';
//import PropTypes from 'prop-types';
import RecipeRow from './RecipeRow';

const RecipeList = ({recipes, onRecipeRowClick, selectedRecipeId}) => {
  return (
    <ul>
      {Object.keys(recipes).map(key => (
        <RecipeRow
          key={key}
          title={recipes[key].title}
          selected={selectedRecipeId === key}
          onClick={() => onRecipeRowClick(key)}
        />
      ))}
    </ul>
  )
};

// TODO: Add RecipeList.propTypes

export default RecipeList;
