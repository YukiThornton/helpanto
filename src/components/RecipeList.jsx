import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'material-ui/List';
import TextField from 'material-ui/TextField';
import Spinner from 'react-spinkit';
// TODO: Move the button away
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';

import RecipeRow from './RecipeRow';
import '../styles/App.css';

class RecipeList extends Component {
    static propTypes = {
      isFetching: PropTypes.bool.isRequired,
      recipes: PropTypes.oneOfType([
        PropTypes.shape({
          kind: PropTypes.string.isRequired,
          title: PropTypes.string.isRequired,
          content: PropTypes.shape({
            memo: PropTypes.string.isRequired,
          }),
          createdAt: PropTypes.string.isRequired,
          lastModifiedAt: PropTypes.string.isRequired,
        }),
        PropTypes.shape({}),
      ]).isRequired,
      selectedRecipeId: PropTypes.string.isRequired,
      onRecipeRowClick: PropTypes.func.isRequired,
      fetchRecipes: PropTypes.func.isRequired,
      openModalNewRecipe: PropTypes.func.isRequired,
      onSearchInputChange: PropTypes.func.isRequired,
      deselectRecipe: PropTypes.func.isRequired,
      onClickRecipeDeleteBtn: PropTypes.func.isRequired,
    }

    componentDidMount() {
      this.props.fetchRecipes();
    }

    onSearchInputChange = (searchText) => {
      this.props.onSearchInputChange(searchText);
    };

    onClickRecipeDeleteBtn = (id) => {
      if (id === this.props.selectedRecipeId) {
        this.props.deselectRecipe();
      }
      this.props.onClickRecipeDeleteBtn(id);
    };

    render() {
      if (this.props.isFetching) {
        return (
          <div className="Spinner">
            <Spinner name="chasing-dots" color="gray" />
          </div>
        );
      }
      return (
        <div className="Recipe-list">
          <TextField
            hintText="Search"
            style={{
              paddingLeft: 10,
            }}
            onChange={(e, val) => this.onSearchInputChange(val)}
          />
          <List>
            {Object.keys(this.props.recipes).map(key => (
              <RecipeRow
                key={key}
                id={key}
                title={this.props.recipes[key].title}
                selected={this.props.selectedRecipeId === key}
                onClick={() => this.props.onRecipeRowClick(key)}
                onClickDeleteBtn={id => this.onClickRecipeDeleteBtn(id)}
              />
            ))}
          </List>
          <FloatingActionButton
            onClick={this.props.openModalNewRecipe}
            style={{
              position: 'fixed',
              top: 50,
              right: 30,
            }}
          >
            <ContentAdd />
          </FloatingActionButton>
        </div>
      );
    }
}

export default RecipeList;
