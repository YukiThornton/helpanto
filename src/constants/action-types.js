/* error-actions.js */
export const ENQUEUE_ERROR = 'ENQUEUE_ERROR';
export const DEQUEUE_ERROR = 'DEQUEUE_ERROR';

/* modal-actions.js */
export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';

/* recipe-actions.js */
export const SELECT_RECIPE_ON_LIST = 'SELECT_RECIPE_ON_LIST';
export const DESELECT_RECIPE_ON_LIST = 'DESELECT_RECIPE_ON_LIST';
export const FILTER_RECIPE_LIST = 'FILTER_RECIPE_LIST';
export const REQUEST_GET_RECIPES = 'REQUEST_GET_RECIPES';
export const REQUEST_GET_RECIPES_SUCCESS = 'REQUEST_GET_RECIPES_SUCCESS';
export const REQUEST_GET_RECIPES_FAILURE = 'REQUEST_GET_RECIPES_FAILURE';
export const REQUEST_POST_RECIPE = 'REQUEST_POST_RECIPE';
export const REQUEST_POST_RECIPE_SUCCESS = 'REQUEST_POST_RECIPE_SUCCESS';
export const REQUEST_POST_RECIPE_FAILURE = 'REQUEST_POST_RECIPE_FAILURE';
export const REQUEST_DELETE_RECIPE = 'REQUEST_DELETE_RECIPE';
export const REQUEST_DELETE_RECIPE_SUCCESS = 'REQUEST_DELETE_RECIPE_SUCCESS';
export const REQUEST_DELETE_RECIPE_FAILURE = 'REQUEST_DELETE_RECIPE_FAILURE';
