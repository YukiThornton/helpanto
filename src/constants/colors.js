export const BLUE_GRAY_100 = '#CFD8DC';
export const BLUE_GRAY_300 = '#90A4AE';
export const BLUE_GRAY_400 = '#78909C';
export const BLUE_GRAY_700 = '#455A64';
