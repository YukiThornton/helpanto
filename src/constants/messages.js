export const MSG_ERR_FETCH_RECIPES = 'Failed to get recipe lists. See you later.';
export const MSG_ERR_CREATE_RECIPE = 'Failed to create a new recipe. See you later.';
export const MSG_ERR_DELETE_RECIPE = 'Failed to delete a recipe. See you later.';
