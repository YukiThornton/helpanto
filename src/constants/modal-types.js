export const MODAL_TYPE_NEW_RECIPE = 'MODAL_TYPE_NEW_RECIPE';
export const MODAL_TYPE_NONE = 'MODAL_TYPE_NONE';
