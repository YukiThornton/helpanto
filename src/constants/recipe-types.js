export const RECIPE_TYPE_MEMO = 'RECIPE_TYPE_MEMO';
export { RECIPE_TYPE_MEMO as default };
