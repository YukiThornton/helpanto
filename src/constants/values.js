export const INVALID_RECIPE_ID = '-1';
export { INVALID_RECIPE_ID as default };
