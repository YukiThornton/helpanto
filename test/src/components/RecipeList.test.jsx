import React from 'react';
import { shallow } from 'enzyme';
import RecipeList from '../../../src/components/RecipeList';
import { INVALID_RECIPE_ID } from '../../../src/constants/values';
import * as recipeTypes from '../../../src/constants/recipe-types';

describe('test RecipeList', () => {
  const recipes = {
    id0: {
      recipeType: recipeTypes.RECIPE_TYPE_MEMO,
      title: 'title1',
      body: {
        memo: 'content1',
      },
      createdAt: '2017-12-01T06:44:04.262Z',
      lastModifiedAt: '2017-12-01T06:44:04.262Z',
    },
    id1: {
      recipeType: recipeTypes.RECIPE_TYPE_MEMO,
      title: 'title2',
      body: {
        memo: 'content2',
      },
      createdAt: '2017-12-02T06:44:04.262Z',
      lastModifiedAt: '2017-12-02T06:44:04.262Z',
    },
  };

  it('should render a spinner correctly when fetching === true', () => {
    const component = (
      <RecipeList
        isFetching
        recipes={recipes}
        selectedRecipeId="id1"
        onRecipeRowClick={() => {}}
        fetchRecipes={() => {}}
        openModalNewRecipe={() => {}}
        onSearchInputChange={() => {}}
        onClickRecipeDeleteBtn={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });

  it('should render a list with 2 items (1 selected) correctly', () => {
    const component = (
      <RecipeList
        isFetching={false}
        recipes={recipes}
        selectedRecipeId="id1"
        onRecipeRowClick={() => {}}
        fetchRecipes={() => {}}
        openModalNewRecipe={() => {}}
        onSearchInputChange={() => {}}
        onClickRecipeDeleteBtn={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });

  it('should render a list with 2 items (0 selected) correctly', () => {
    const component = (
      <RecipeList
        isFetching={false}
        recipes={recipes}
        selectedRecipeId={INVALID_RECIPE_ID}
        onRecipeRowClick={() => {}}
        fetchRecipes={() => {}}
        openModalNewRecipe={() => {}}
        onSearchInputChange={() => {}}
        onClickRecipeDeleteBtn={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });

  it('should render a list with 0 items correctly', () => {
    const component = (
      <RecipeList
        isFetching={false}
        recipes={{}}
        selectedRecipeId={INVALID_RECIPE_ID}
        onRecipeRowClick={() => {}}
        fetchRecipes={() => {}}
        openModalNewRecipe={() => {}}
        onSearchInputChange={() => {}}
        onClickRecipeDeleteBtn={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });
});
