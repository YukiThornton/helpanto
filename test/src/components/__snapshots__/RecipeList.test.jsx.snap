// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RecipeList should render a list with 0 items correctly 1`] = `
<div
  className="Recipe-list"
>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Search"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    style={
      Object {
        "paddingLeft": 10,
      }
    }
    type="text"
    underlineShow={true}
  />
  <List />
  <FloatingActionButton
    disabled={false}
    mini={false}
    onClick={[Function]}
    secondary={false}
    style={
      Object {
        "position": "fixed",
        "right": 30,
        "top": 50,
      }
    }
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
</div>
`;

exports[`test RecipeList should render a list with 2 items (0 selected) correctly 1`] = `
<div
  className="Recipe-list"
>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Search"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    style={
      Object {
        "paddingLeft": 10,
      }
    }
    type="text"
    underlineShow={true}
  />
  <List>
    <RecipeRow
      id="id0"
      key="id0"
      onClick={[Function]}
      onClickDeleteBtn={[Function]}
      selected={false}
      title="title1"
    />
    <RecipeRow
      id="id1"
      key="id1"
      onClick={[Function]}
      onClickDeleteBtn={[Function]}
      selected={false}
      title="title2"
    />
  </List>
  <FloatingActionButton
    disabled={false}
    mini={false}
    onClick={[Function]}
    secondary={false}
    style={
      Object {
        "position": "fixed",
        "right": 30,
        "top": 50,
      }
    }
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
</div>
`;

exports[`test RecipeList should render a list with 2 items (1 selected) correctly 1`] = `
<div
  className="Recipe-list"
>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Search"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    style={
      Object {
        "paddingLeft": 10,
      }
    }
    type="text"
    underlineShow={true}
  />
  <List>
    <RecipeRow
      id="id0"
      key="id0"
      onClick={[Function]}
      onClickDeleteBtn={[Function]}
      selected={false}
      title="title1"
    />
    <RecipeRow
      id="id1"
      key="id1"
      onClick={[Function]}
      onClickDeleteBtn={[Function]}
      selected={true}
      title="title2"
    />
  </List>
  <FloatingActionButton
    disabled={false}
    mini={false}
    onClick={[Function]}
    secondary={false}
    style={
      Object {
        "position": "fixed",
        "right": 30,
        "top": 50,
      }
    }
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
</div>
`;

exports[`test RecipeList should render a spinner correctly when fetching === true 1`] = `
<div
  className="Spinner"
>
  <Spinner
    color="gray"
    fadeIn="full"
    name="chasing-dots"
    noFadeIn={false}
    overrideSpinnerClassName=""
  />
</div>
`;
